### PR TITLE
Cancel `ServerHandler` when `ServerTransportConnection.incomingStreams` is closed  

### PR DESCRIPTION
**Pull request by sigurdm**
I take the chance to also deprecate ServerHandler and Server.serveStream.
This should fix: https://github.com/grpc/grpc-dart/issues/147

### CODE REVIEW

**Comment by sigurdm** [comment]
cc @jonasfj @mkustermann @a14n

File: lib/src/server/server.dart [hunk 1]
```diff
@@ -101,20 +102,35 @@ class Server {
...
       // TODO(jakobr): Set active state handlers, close connection after idle
       // timeout.
-      connection.incomingStreams.listen(serveStream, onError: (error) {
+      connection.incomingStreams.listen((stream) {
+        handler = serveStream_(stream);
+      }, onError: (error) {
         print('Connection error: $error');
       }, onDone: () {
+        // TODO(sigurdm): This is not correct behavior in the presence of
+        // half-open tcp streams.
```
Comment (jonasfj) [inline lib/src/server/server.dart]: I think it's better to call them 'half-closed', as half-open is something else... sorry, I might have used the wrong terminology from time to time :)

**Comment by sigurdm** [comment]
OOps. Thanks

File: test/server_handles_broken_connection_test.dart [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  }
+}
+
+class ClientData {
+  final int port;
+  final SendPort sendPort;
+  ClientData({this.port, this.sendPort});
+}
+
+void client(clientData) async {
```
Comment (a14n) [inline test/server_handles_broken_connection_test.dart]: Use `ClientData` as type for `clientData` ?

**Comment by sigurdm** [comment]
Thanks for the help with this @a14n and others!

**Comment by a14n** [comment]
Thank you for the fix!

**Pull request merged**